### DISCUSSION TITLE
使用Google DoT作为上游DNS，日志大量报错Bad address

### DISCUSSION THREAD

**Discussion by ghost** [pymumu/smartdns#1408] [Show and tell]
试过开/关 TLS主机名校验、SNI名称校验、SPKI都没有用。

/tmp/log/smartdns/smartdns.log 日志大量报错：
[ERROR][     dns_client.c:2635] recv failed, server 8.8.8.8:853, Bad address
[ERROR][     dns_client.c:2635] recv failed, server 8.8.4.4:853, Bad address

然而这两个IP都能ping通，UDP请求、DoH也都没问题，一旦用了DoT日志就会报错。
类型的配置Cloudflare、OpenDNS、Quad9的DoT都没有问题，就Google的不行。

不知道各位有同样的情况么？

**Comment by Benjamin1919** [comment]
我用的是DOH，也出现了这样的报错：

[2023-12-08 16:48:37,531][ERROR][     dns_client.c:2635] recv failed, server 8.8.8.8:443, Bad address
[2023-12-08 17:49:32,435][ERROR][     dns_client.c:2635] recv failed, server 8.8.8.8:443, Bad address

几乎每隔一小时就出现一次，加不加socks代理都是一样的。

如果有知道解决方案的麻烦不吝赐教

**Reply by PikuZheng** [reply]
证实一下用的是哪个版本，不要用43.0.35至43.0.38

**Reply by Benjamin1919** [reply]
openwrt里看到的版本号是Release43-27

**Reply by PikuZheng** [reply]
不知道这是哪个版本啊，要么降级到release里的43，要么升级到最新的吧

**Comment by pymumu** [comment]
最新代码看看，有问题发下log。
